.github/workflows/ci.yml

```diff
@@ -0,0 +1,84 @@
+---
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Python CI
+
+# yamllint disable-line rule:truthy
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  CI-Python:
+    runs-on: ubuntu-latest
+    env:
+      PYTHONDEVMODE: 1
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.11", "3.12", "3.13"]
+        # Empty is latest, head is latest from GitHub
+        pdm-version: [""]
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: pdm-project/setup-pdm@main
+        name: Setup PDM
+        with:
+          cache: true
+          python-version: ${{ matrix.python-version }}             # Version range or exact version of a Python version to use, the same as actions/setup-python
+          architecture: x64                # The target architecture (x86, x64) of the Python interpreter. the same as actions/setup-python
```

Comment:
Can we add x86 & arm 32/64 from the first release, for mac MX users ? :pray:

Reply:
We don't need it, as our code is pure Python it is universal. We could expand the architecture/OS matrix if we want, but as there are no tests yet it would not be useful.
The Docker image is arch specific, but already has amd64 and arm64 as targets.